client/src/components/Overview/Overview.jsx

```diff
@@ -1 +1,54 @@
-import React from 'react';
+/* eslint-disable import/extensions */
+import React, { useState, useContext, useEffect } from 'react';
+import axios from 'axios';
+import ProductContext from '../../context';
+import ImageGallery from './ImageGallery.jsx';
+import ProductDetails from './ProductDetails.jsx';
+import FeaturesContainer from './FeaturesContainer.jsx';
+
+export default function Overview() {
+  const { products } = useContext(ProductContext);
+  const [currentProduct, setCurrentProduct] = useState({});
+  const [styles, setStyles] = useState([]);
+
+  function fetchStyles() {
+    axios.get(`/products/${products[0].id}/styles`).then((result) => result.data.results).then(setStyles);
+  }
+
+  function fetchCurrentProduct() {
+    axios
+      .get(`/products/${products[0].id}`)
+      .then((result) => result.data)
+      .then(setCurrentProduct);
+  }
+
+  useEffect(() => {
+    fetchCurrentProduct();
+  }, []);
+
+  useEffect(() => {
+    fetchStyles();
+  }, []);
+
+  if (styles.length === 0) {
+    return (
+      <div>Loading...</div>
+    );
+  }
+  return (
+    <ProductContext.Provider value={{
```

Comment:
It feels confusing to have two different ProductContext.Providers with different shapes.  Should we just have one?

Reply:
I was unsure about that. If we create a new state in a lower file, as I did here in Overview, should we just create a new context to pass stuff down? Or have all state stored in App, and then pass down accordingly? May need to utilize Promises (probably async await) if we choose to do the latter, so we can avoid errors as we try to use data deeper.